samza-api/src/main/java/org/apache/samza/operators/windows/Windows.java

```diff
@@ -16,85 +16,327 @@
   }
 
   /**
-   * Static API method to create a {@link SessionWindow} in which the output value is a collection of {@code SI} from the input {@link MessageEnvelope}s
-   *
-   * @param sessionKeyFunction  function to calculate session window key
-   * @param sessionInfoExtractor  function to retrieve session info of type {@code SI} from the input {@link MessageEnvelope} of type {@code M}
-   * @param <M>  type of the input {@link MessageEnvelope}
-   * @param <WK>  type of the session window key
-   * @param <SI>  type of the session information retrieved from each input {@link MessageEnvelope} of type {@code M}
-   * @return  the {@link Window} function for the session
+   * Creates a {@link Window} that groups incoming {@link MessageEnvelope}s into fixed-size, non-overlapping
+   * processing time based windows.
+   *
+   * <p>The example below windows the stream into fixed-size 10 second windows, and computes a windowed-percentile.
+   *
+   * <pre> {@code
+   * MessageStream<Long> stream = ...;
+   * Function<Collection<Long, Long>> percentile99 = ..
+   *
+   * MessageStream<WindowOutput<WindowKey, Collection<Long>>> windowedStream = integerStream.window(Windows.tumblingWindow(Duration.seconds(10)));
+   * MessageStream<Long> windowedPercentiles = windowed.map(windowedOutput -> percentile99(windowedOutput.getMessage());
+   * }
+   * </pre>
+   *
+   * @param duration the duration in processing time.
+   * @param <M> the type of the input {@link MessageEnvelope}
+   * @return the created {@link Window} function.
    */
-  public static <M extends MessageEnvelope, WK, SI> Window<M, WK, Collection<SI>, WindowOutput<WK, Collection<SI>>> intoSessions(Function<M, WK> sessionKeyFunction,
-      Function<M, SI> sessionInfoExtractor) {
-    return new SessionWindow<>(sessionKeyFunction, (m, c) -> {
-        c.add(sessionInfoExtractor.apply(m));
-        return c;
-      }
-    );
+  public static <M extends MessageEnvelope> Window<M, Void, WindowKey<Void>, Collection<M>, WindowOutput<WindowKey<Void>, Collection<M>>> tumblingWindow(Duration duration) {
+    BiFunction<M, Collection<M>, Collection<M>> aggregator = (m, c) -> {
+      c.add(m);
+      return c;
+    };
+    return tumblingWindow(duration, aggregator);
   }
 
   /**
-   * Static API method to create a {@link SessionWindow} as a counter of input {@link MessageEnvelope}s
+   * Creates a {@link Window} that groups incoming {@link MessageEnvelope}s into sessions per-key based on the provided {@code sessionGap}
+   * and applies the provided aggregate function to them.
+   *
+   * <p>A <i>session</i> captures some period of activity over a {@link org.apache.samza.operators.MessageStream}.
+   * A session is considered complete when no new messages arrive within the {@code sessionGap}. All {@link MessageEnvelope}s that arrive within
+   * the gap are grouped into the same session.
+   *
+   * <p>The example below computes the maximum value per-key over a session window of gap 10 seconds.
```

Comment:
Still prefer p, li tags etc on their own lines for readability when looking at docs directly (as opposed to a javadoc browser).